action.yml

```diff
@@ -0,0 +1,40 @@
+name: 'Upload mender artifact'
+description: 'Upload a mender artifact to a hosted Mender backend instance'
+inputs:
+  mender_pat:
+    description: 'personal access token for the hosted Mender backend account to be used'
+    required: true
+  mender_artifact:
+    description: 'path of the artifact to be uploaded, relative to GITHUB_WORKSPACE'
+    required: true
+  mender_uri:
+    description: 'URI for the hosted Mender backend account to be used'
+    required: false
+    default: 'https://hosted.mender.io'
+runs:
+  using: "composite"
+  steps:
+    - id: check_file
+      shell: bash
+      run: |
+        ls ${GITHUB_WORKSPACE}/${{ inputs.mender_artifact }} > /dev/null
+    - id: upload_artifact
+      shell: bash
+      run: |
+        RESPONSE=$(curl -s -X POST ${{ inputs.mender_uri }}/api/management/v1/deployments/artifacts \
+          -H "Content-Type: multipart/form-data" \
+          -H "Accept: application/json" \
+          -H "Authorization: Bearer ${{ inputs.mender_pat }}" \
+          -F artifact=@${GITHUB_WORKSPACE}/${{ inputs.mender_artifact }})
+        if [[ ! -z "$RESPONSE" || "$RESPONSE" != "" ]]; then
+          if $(echo ${RESPONSE} | grep -q "An artifact with the same name has conflicting depends"); then
+            echo "WARN: artifact ${{ inputs.mender_artifact }} already exists on the server ${{ inputs.mender_uri }}, upload is skipped"
+            echo "Server's response: ${RESPONSE}"
+            exit 0
+          else
+            echo "ERROR: failed to upload artifact ${{ inputs.mender_artifact }} to ${{ inputs.mender_uri }}"
+            echo "Server's response: ${RESPONSE}"
+            exit 1
+          fi
+        fi
+        echo "INFO: artifact ${{ inputs.mender_artifact }} successfully uploaded to ${{ inputs.mender_uri }}"
```

Comment:
Where is the Docker image set in this file? I cannot see a pointer to `mendersoftware/mender-ci-tools`

Reply:
Nowhere. This action doesn't depend from it, it depends only from `bash` and `curl`. See previous comment.